### PR TITLE
docs: add docs about API, create new pages

### PR DESCRIPTION
**Pull request by layershifter**
This PR adds documentation for existing API, pages about SSR and child window rendering.

### CODE REVIEW

File: apps/website/docs/react/api/create-dom-renderer.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+sidebar_position: 4
+---
+
+# createDOMRenderer
+
+`createDOMRenderer` is paired with `RendererProvider` component and is useful for [child window rendering](/react/guides/child-window-rendering) and [SSR](/react/guides/ssr-usage) scenarios. It helps to ensure that styles will be inserted to a proper document.
```
Comment (ling1726) [inline apps/website/docs/react/api/create-dom-renderer.md:7]: should we mention that it' s possible to create custom renderers ? you can use the one in the playground as reference

Comment (ling1726) [inline apps/website/docs/react/api/create-dom-renderer.md:7]:
```suggestion
`createDOMRenderer` is paired with `RendererProvider` component and is useful for [child window rendering](/react/guides/child-window-rendering) and [SSR](/react/guides/ssr-usage) scenarios. This is the default renderer for web, and will make sure that styles are injected to a document.
```

File: apps/website/docs/react/api/make-static-styles.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+---
+sidebar_position: 3
+---
+
+import OutputTitle from '@site/src/components/OutputTitle'
+
+# makeStaticStyles
+
+Creates styles attached to a global selector. This is especially useful for CSS resets, for example [normalize.css](https://github.com/necolas/normalize.css/).
```
Comment (ling1726) [inline apps/website/docs/react/api/make-static-styles.md:9]:
```suggestion
Creates styles with a global selector. This is especially useful for CSS resets, for example [normalize.css](https://github.com/necolas/normalize.css/).
```

File: apps/website/docs/react/api/merge-classes.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
+---
+sidebar_position: 2
+---
+
+import OutputTitle from '@site/src/components/OutputTitle'
+
+# mergeClasses
+
+There are cases where you need to merge classes from multiple `useClasses` calls. To properly merge the classes, you need to use `mergeClasses()` function, which performs merge and deduplication of atomic classes generated by `makeStyles()`.
```
Comment (ling1726) [inline apps/website/docs/react/api/merge-classes.md:9]:
```suggestion
There are cases where you need to merge classes from multiple `useClasses` calls. To merge the classes correctly, you need to use `mergeClasses()` function, which merges and dedupes the atomic classes generated by `makeStyles()`.
```

File: apps/website/docs/react/api/merge-classes.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      <div className={firstClassName} />
+      <div className={secondClassName} />
+    </>
+  );
+}
+```
+
+:::note
+
+Compared to native CSS order of arguments in `mergeClasses()` determines the results. This allows to control styles with JavaScript.
```
Comment (ling1726) [inline apps/website/docs/react/api/merge-classes.md:46]:
```suggestion
Compared to native CSS order of arguments in `mergeClasses()` determines the results. This lets you control styles with JavaScript.
```

File: apps/website/docs/react/api/merge-classes.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+## Incorrect usages
+
+This section shows and describes anti-patterns which should never be used.
+
+### Never concatenate class strings
+
+:::caution
+
+It is not possible to simply concatenate classes returned by `useClasses()` hooks. Always use `mergeClasses()` to perform classes merging.
```
Comment (ling1726) [inline apps/website/docs/react/api/merge-classes.md:58]:
```suggestion
It is not possible to simply concatenate classes returned by `useClasses()` hooks. Always use `mergeClasses()` to merge classes.
```

File: apps/website/docs/react/api/merge-classes.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+:::
+
+## Incorrect usages
+
+This section shows and describes anti-patterns which should never be used.
+
+### Never concatenate class strings
+
+:::caution
```
Comment (ling1726) [inline apps/website/docs/react/api/merge-classes.md:56]: Let' s also give a code sample for merging without `mergeClasses`

File: apps/website/docs/react/api/merge-classes.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+function App(props) {
+  const classes = useClasses();
+  // ⚠️ Never concatenate class strings, always use mergeClasses()
+  const wrongClasses = classes.rootA + ' ' + classes.rootB;
+}
+```
+
+### Order of arguments matters
+
+There might be an issue with incorrect order of classes - as the latest wins. In the example below `props.className` should be last to override the component styles:
```
Comment (ling1726) [inline apps/website/docs/react/api/merge-classes.md:77]:
```suggestion
The order of classes is important - If two classes apply the same style, the latest class wins. In the example below the background color will be red:
```

Let's not use a library component specific example, simple red/blue will be much clearer

File: apps/website/docs/react/guides/child-window-rendering.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
+---
+sidebar_position: 5
+---
+
+# Child Window/Shadow DOM Rendering
+
+When rendering on the main browser window, many components need access to window or document for applying styling, listening for events, or measuring things. However it is possible to render to child windows and elements hosted in iframe elements.
```
Comment (ling1726) [inline apps/website/docs/react/guides/child-window-rendering.md:7]:
```suggestion
When rendering in the main browser window, many components will need access to window or document to apply styles, listening for events, or measuring things. However it is possible to render to child windows and elements hosted in iframes.
```

File: apps/website/docs/react/guides/child-window-rendering.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+In these cases, the target element is hosted in a different context, and thus have a different window reference. To aid in providing components with the correct instances of window or document, React context can be used to provide the tree of React components with the correct instance.
```
Comment (ling1726) [inline apps/website/docs/react/guides/child-window-rendering.md:9]:
```suggestion
```

I would completely rework this, it's like reading chinese for me 😆

File: apps/website/docs/react/guides/child-window-rendering.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+## Configuring rendering
```
Comment (ling1726) [inline apps/website/docs/react/guides/child-window-rendering.md:11]:
```suggestion
## Configure rendering
```

File: apps/website/docs/react/guides/child-window-rendering.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+  return (
+    <RendererProvider renderer={renderer} targetDocument={targetDocument}>
+      {children}
+    </RendererProvider>
+  );
+}
+```
+
+You can check complete example at [CodeSandbox](https://codesandbox.io/s/griffel-react-rendering-into-iframe-btezpu).
```
Comment (ling1726) [inline apps/website/docs/react/guides/child-window-rendering.md:31]:
```suggestion
You can check out the complete example in [CodeSandbox](https://codesandbox.io/s/griffel-react-rendering-into-iframe-btezpu).
```

File: apps/website/docs/react/guides/ssr-usage.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
+---
+sidebar_position: 4
+---
+
+# Server-Side Rendering
+
+Griffel fully supports Server-Side Rendering.
```
Comment (ling1726) [inline apps/website/docs/react/guides/ssr-usage.md:7]:
```suggestion
Griffel provides first class support for Server-Side Rendering.
```

File: apps/website/docs/react/guides/ssr-usage.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+## Next.js
+
+### Base setup
+
+For basic instructions on getting Next.js set up, see [Getting Started](https://nextjs.org/docs/getting-started). Please complete following steps:
```
Comment (ling1726) [inline apps/website/docs/react/guides/ssr-usage.md:13]:
```suggestion
For basic instructions to setup Next.js, see [Getting Started](https://nextjs.org/docs/getting-started). Please complete following steps:
```

File: apps/website/docs/react/guides/ssr-usage.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+class MyDocument extends Document {
+  static async getInitialProps(ctx) {
+    // 👇 creates a renderer
+    const renderer = createDOMRenderer();
+    const originalRenderPage = ctx.renderPage;
+
+    ctx.renderPage = () =>
+      originalRenderPage({
+        // 👇 this is required by pass the same instance to <App />
```
Comment (ling1726) [inline apps/website/docs/react/guides/ssr-usage.md:36]:
```suggestion
```

what does this sentence mean ?

File: apps/website/docs/react/guides/ssr-usage.md [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+        </body>
+      </Html>
+    );
+  }
+}
+
+export default MyDocument;
+```
+
+2. Create or modify a `_app.js` file under your `pages` folder with the following content:
```
Comment (ling1726) [inline apps/website/docs/react/guides/ssr-usage.md:66]:
```suggestion
2. Create or modify an `_app.js` file under your `pages` folder with the following content:
```

File: apps/website/docs/react/guides/ssr-usage.md [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+
+```js
+import { createDOMRenderer, RendererProvider } from '@griffel/react';
+
+function MyApp({ Component, pageProps, renderer }) {
+  return (
+    // 👇 accepts a renderer passed from <Document /> component or creates a default one
```
Comment (ling1726) [inline apps/website/docs/react/guides/ssr-usage.md:73]:
```suggestion
    // 👇 accepts a renderer passed from the <Document /> component or creates a default one
```

File: apps/website/docs/react/install.mdx [hunk 17]
```diff
@@ -25,3 +25,5 @@ npm install @griffel/react
 
 </TabItem>
 </Tabs>
+
+For quick start you can use [CodeSandbox template](https://codesandbox.io/s/griffel-react-template-p5e8mh) or check our [try out feature](/griffel/try-it-out).
```
Comment (ling1726) [inline apps/website/docs/react/install.mdx:29]:
```suggestion
For quick start you can use [CodeSandbox template](https://codesandbox.io/s/griffel-react-template-p5e8mh) or check our [try it out feature](/griffel/try-it-out).
```

File: apps/website/docs/react/api/merge-classes.md (hunk 6 above)
Comment (layershifter) [reply apps/website/docs/react/api/merge-classes.md:56]: Not sure that I get this, we have a sample below that shows wrong behavior:

image

Comment (ling1726) [reply apps/website/docs/react/api/merge-classes.md:56]: I guess, my point is that currently the section is just saying "dont't merge classes without `mergeClasses`".

I'm missing the why

**Review by ling1726 — APPROVED** [review]

File: apps/website/docs/react/api/create-dom-renderer.md (hunk 1 above)
Comment (layershifter) [reply apps/website/docs/react/api/create-dom-renderer.md]: Not now probably as I am not entirely sure about the API. For example, what happens when we will have CSS extraction?

File: apps/website/docs/react/api/merge-classes.md (hunk 6 above)
Comment (layershifter) [reply apps/website/docs/react/api/merge-classes.md:56]: Added a note why.

**Review by ling1726 — APPROVED** [review]

**Pull request merged**